### PR TITLE
Small fixes in the form parameters tests

### PR DESCRIPTION
**Pull request by mihalimara22**
Fixes:
- mistake in error output
- missing error checking when parsing forms

### CODE REVIEW

File: tests/formparams/form_params_test.go [hunk 1]
```diff
@@ -41,6 +41,10 @@ func TestSimpleFormParameters(t *testing.T) {
 		"\r\n")
 	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
 		req.ParseForm()
+		err := req.ParseForm()
```
Comment (mattiasgrenfeldt) [inline tests/formparams/form_params_test.go:44]: Double call to `req.ParseForm()`. Remove the old one

File: tests/formparams/form_params_test.go [hunk 2]
```diff
@@ -143,7 +147,10 @@ func TestFormParametersValidDuplicateContentLength(t *testing.T) {
 		if diff := cmp.Diff(want, got); diff != "" {
 			t.Errorf("req.Header: got %v, want %v, diff (-want +got): \n%s", got, want, diff)
 		}
-		req.ParseForm()
+		err := req.ParseForm()
+		if err != nil {
+			t.Errorf("req.ParseForm: got %v, want nil", err)
```
Comment (mattiasgrenfeldt) [inline tests/formparams/form_params_test.go:152]: Write `req.ParseForm()` to be consistent

Comment (mattiasgrenfeldt) [reply tests/formparams/form_params_test.go:151]: You forgot to change this here.

File: tests/formparams/form_params_test.go [hunk 3]
```diff
@@ -40,7 +40,10 @@ func TestSimpleFormParameters(t *testing.T) {
 		reqBody + "\r\n" +
 		"\r\n")
 	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
-		req.ParseForm()
+		err := req.ParseForm()
+		if err != nil {
```
Comment (empijei) [inline tests/formparams/form_params_test.go:44]:
```suggestion
		if err := req.ParseForm(); err != nil {
```
Same below.

File: tests/formparams/form_params_test.go [hunk 4]
```diff
@@ -213,16 +219,21 @@ func TestFormParametersBreakUrlEncoding(t *testing.T) {
 		"veggie=%sc&fruit=apple\r\n" +
 		"\r\n")
 	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
-		req.ParseForm()
+		err := req.ParseForm().Error()
+		wantErr := `invalid URL escape "%sc"`
+		if err != wantErr {
+			t.Errorf(`req.ParseForm(): want %v, got %v`, wantErr, err)
+		}
+
```
Comment (kele) [inline tests/formparams/form_params_test.go:227]: Avoid comparing full error strings, as this makes the tests essentially "change detectors". In this case, it is fine to just check whether the error is non-nil.

File: tests/formparams/form_params_test.go [hunk 5]
```diff
@@ -213,16 +219,21 @@ func TestFormParametersBreakUrlEncoding(t *testing.T) {
...
 		got := req.Form["veggie"]
 		if diff := cmp.Diff(want, got); diff != "" {
-			t.Errorf(`URL.Query()["veggie"]:, got %v, want %v`, req.Form["vegetable"], nil)
+			t.Errorf(`Form["veggie"]:, got %v, want %v`, req.Form["vegetable"], nil)
 		}
 		want = []string{"apple"}
 		got = req.Form["fruit"]
 		if diff := cmp.Diff(want, got); diff != "" {
-			t.Errorf(`URL.Query()["fruit"]:, got %v, want %v`, req.Form["fruit"], want)
+			t.Errorf(`Form["fruit"]:, got %v, want %v`, req.Form["fruit"], want)
```
Comment (kele) [inline tests/formparams/form_params_test.go:236]: nit: drop the comma

File: tests/formparams/form_params_test.go [hunk 6]
```diff
@@ -213,16 +219,21 @@ func TestFormParametersBreakUrlEncoding(t *testing.T) {
 		"veggie=%sc&fruit=apple\r\n" +
 		"\r\n")
 	resp, err := requesttesting.MakeRequest(context.Background(), postReq, func(req *http.Request) {
-		req.ParseForm()
+		err := req.ParseForm().Error()
+		wantErr := `invalid URL escape "%sc"`
+		if err != wantErr {
+			t.Errorf(`req.ParseForm(): want %v, got %v`, wantErr, err)
```
Comment (kele) [inline tests/formparams/form_params_test.go:225]: nit: got before want

File: tests/formparams/form_params_test.go (hunk 5 above)
Comment (mihalimara22) [reply tests/formparams/form_params_test.go]: Resolved in [a180754](url)

File: tests/formparams/form_params_test.go (hunk 6 above)
Comment (mihalimara22) [reply tests/formparams/form_params_test.go]: Resolved in [a180754](url)

File: tests/formparams/form_params_test.go (hunk 4 above)
Comment (mihalimara22) [reply tests/formparams/form_params_test.go]: Resolved in [a180754](url)

File: tests/formparams/form_params_test.go (hunk 3 above)
Comment (mihalimara22) [reply tests/formparams/form_params_test.go]: Resolved in [a180754](url)

**Pull request merged**